dataprofiler/profilers/profile_builder.py

```diff
@@ -483,16 +482,20 @@ def report(self, report_options=None):
         return _prepare_report(report, output_format, omit_keys)
 
     def _get_unique_row_ratio(self):
-        return len(self.hashed_row_dict) / self.rows_ingested
+        return len(self.hashed_row_dict) / self.total_samples
 
     def _get_row_is_null_ratio(self):
-        return self.row_is_null_count / self.rows_ingested
+        columns = list(self._profile.values())
+        samples_used = columns[0].sample_size
+        return self.row_is_null_count / samples_used
 
     def _get_row_has_null_ratio(self):
-        return self.row_has_null_count / self.rows_ingested
+        columns = list(self._profile.values())
+        samples_used = columns[0].sample_size
```

Comment:
this value needs to be the max of all `column[0].sample_size` otherwise the ratio could be > 1